### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Michelle219**

### CODE REVIEW

File: md_client/client.py [hunk 1]
```diff
@@ -36,17 +42,36 @@ async def sync_with_remote(self):
...
+            logging.info("DBs are not synced! Syncing dbs, this might overwrite data")
+            db_data = md_pb2.GetDB()
+            db_data.ParseFromString(await self.reader.read())
+            with open(self.db_name, 'wb') as f:
+                f.write(db_data.db_file)
+
+
+def send_protobuf(self, protobuf):
+    logging.debug(f"Sent protobuf {protobuf}")
+    self.writer.write(protobuf)
```
Comment (dean-levinson) [inline md_client/client.py:67]: Maybe we should call async writer.drain after write? When I tried without drain it caused me problems

File: md_server/session.py [hunk 2]
```diff
@@ -1,26 +1,57 @@
...
-    def push_db(self):
-        pass
+    async def get_db_info(self):
+        db_info = md_pb2.DBInfo()
+        db_info.ParseFromString(await self.reader.read())
+        self.client_id = db_info.client_id
+        self.db_name = db_info.db_name
+        logging.info(f"Client {self.client_id} connected to db {self.db_name}")
+
+    async def push_db(self):
```
Comment (dean-levinson) [inline md_server/session.py:35]: A little bit messy but for now it is good

File: mdlib/db_utils.py [hunk 3]
```diff
@@ -1,24 +1,67 @@
...
+        # No need to update other client because this is a new db
+        with open(self.db_path, 'w') as db_descriptor:
+            json.dump({}, db_descriptor)
 
     def handle_protobuf(self, protobuf_obj):
         # call DBProtocol to parse obj
         # call switch case
+        action = DBAction()
+        action.ParseFromString(protobuf_obj)
+        match action:
```
Comment (dean-levinson) [inline mdlib/db_utils.py:39]: hahaha nice

File: mdlib/socket_utils.py [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import struct
+import asyncio
+import logging
+
+
+class LengthReader(asyncio.StreamReader):
```
Comment (dean-levinson) [inline mdlib/socket_utils.py:6]: I liked!

File: mdlib/socket_utils.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+        return b''
+
+
+class LengthWriter(asyncio.StreamWriter):
+    def __init__(self, writer: asyncio.StreamWriter):
+        self.writer = writer
+
+    def write(self, data):
+        total_size = len(data)
+        pack1 = struct.pack('>I', total_size)
```
Comment (dean-levinson) [inline mdlib/socket_utils.py:24]: pack1?

File: md_client/main.py [hunk 6]
```diff
@@ -9,9 +9,10 @@
 
 async def main():
     hostname = "127.0.0.1"
-    db_name = "mydb"
+    db_name ="mydb"
```
Comment (dean-levinson) [inline md_client/main.py:12]: ?

**Review by dean-levinson — COMMENTED** [review]
Very nice!

File: mdlib/socket_utils.py (hunk 5 above)
Comment (Michelle219) [reply mdlib/socket_utils.py:24]: #stackoverflow..

File: md_client/client.py (hunk 1 above)
Comment (Michelle219) [reply md_client/client.py:67]: done

File: md_server/session.py (hunk 2 above)
Comment (Michelle219) [reply md_server/session.py:35]: this need to be changed to be a protobuf action in ACTIONS
TODO..

**Pull request merged**